jenkins/continuous-integration.jenkins

```diff
@@ -0,0 +1,41 @@
+pipeline {
+  agent {
+    docker { image 'node:10-slim' }
+  }
+  environment {
+    CODECOV_TOKEN = '523afe7a-4620-466f-a26a-d914c18a7c0a'
+  }
+  stages {
+    stage('Build') {
+      steps {
+        sh 'yarn'
+      }
+    }
+    stage('Test') {
+      parallel {
+        stage('Lint') {
+          steps {
+            sh 'yarn js-lint'
+            sh 'yarn typescript-lint'
+          }
+        }
+        stage('Type Checks') {
+          steps {
+            sh 'yarn type-check'
+          }
+        }
+        stage('Unit Tests') {
+          steps {
+            sh 'yarn jest --coverage --verbose --maxWorkers=2' 
```

Comment:
Could this just be `yarn test` (& update `package.json` to run `--coverage`)? It is often useful to be able to run locally the same command that jenkins runs and once they diverge too much it becomes hard to reproduce problems in jenkins

Reply:
i initially had it in the package.json but became annoyed with the coverage files that were being generated, i suppose i could just gitignore them if you would prefer to have the command in the package.json